source/server/http/admin.cc

```diff
@@ -992,5 +995,29 @@ bool AdminImpl::removeHandler(const std::string& prefix) {
   return false;
 }
 
+Http::Code AdminImpl::request(absl::string_view path, const Http::Utility::QueryParams& params,
+                              absl::string_view method, Http::HeaderMap& response_headers,
+                              std::string& body) {
+  AdminFilter filter(*this);
+  Http::HeaderMapImpl request_headers;
+  request_headers.insertMethod().value(method.data(), method.size());
+  filter.decodeHeaders(request_headers, false);
+
+  // TODO(jmarantz): make QueryParams a real class and put this serializer there,
+  // along with proper URL escaping of the name and value.
```

Comment:
Yeah, ++escaping.

Reply:
TBH I think what I'd really like to do is change admin filter callbacks to take a const QueryParam& as an arg, rather than having each one parse it.  Then serializing/parsing will be skipped and escaping will be unnecessary.  Changing TODO. I don't want to tackle that in this PR though; will follow up.